ReactAndroid/src/main/java/com/facebook/react/views/art/ARTSurfaceViewShadowNode.java

```diff
@@ -34,23 +42,46 @@ public boolean isVirtualAnchor() {
   @Override
   public void onCollectExtraUpdates(UIViewOperationQueue uiUpdater) {
     super.onCollectExtraUpdates(uiUpdater);
-    uiUpdater.enqueueUpdateExtraData(getReactTag(), drawOutput());
+    drawOutput();
+    uiUpdater.enqueueUpdateExtraData(getReactTag(), this);
   }
 
-  private Object drawOutput() {
-    // TODO(7255985): Use TextureView and pass Surface from the view to draw on it asynchronously
-    // instead of passing the bitmap (which is inefficient especially in terms of memory usage)
-    Bitmap bitmap = Bitmap.createBitmap(
-        (int) getLayoutWidth(),
-        (int) getLayoutHeight(),
-        Bitmap.Config.ARGB_8888);
-    Canvas canvas = new Canvas(bitmap);
+  private void drawOutput() {
+    if (mSurface == null) {
+      markChildrenUpdatesSeen(this);      
```

Comment:
If I do not markUpdateSeen each child in the entire Surface subtree, then ARTSurface's method onCollectExtraUpdates is called ONLY ONCE right after the app start. As I understood from the react-native sources when a child is updated it is marked "update not seen" (i.e. some ARTShape on a ARTSurface) and also it tries to mark its parent as "update not seen" only if parent doesn't have any pending updates. Otherwise if parent itself is already marked "update not seen", then an update is not propagated further to even more distant ancestors and obviously to the ultimate ancestor ARTSurface. 

That's why I need to mark all ARTSurface's children as "UpdateSeen" even if TextureView's Android.Surface is not ready for drawing.

Or is there a pre-built method in ReactNative for this?